src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersions.java

```diff
@@ -0,0 +1,190 @@
+    this.elements = new HashSet<>(versions);
+  }
+
+  /**
+   * Initializes a set of versions.
+   *
+   * @param versions A number of versions.
+   */
+  public ArtifactVersions(ArtifactVersion... versions) {
+    this(setOf(versions));
+  }
+
+  /**
+   * Get a size of the set.
+   *
+   * @return A number of versions in the set.
+   */
+  public int size() {
+    return elements.size();
+  }
+
+  /**
+   * Check if artifact versions set is empty.
+   *
+   * @return true if empty otherwise false.
+   */
+  public boolean empty() {
+    return elements.isEmpty();
+  }
+
+  /**
+   * Checks if the collection contains one of the other versions.
+   *
+   * @param versions The other versions.
+   * @return True if at least one of the other versions is present in the collection,
+   *          false otherwise.
+   */
+  public boolean containsAnyOf(ArtifactVersions versions) {
+    for (ArtifactVersion version : versions) {
+      if (this.elements.contains(version)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Get versions sorted by date.
+   * First entry is the latest release.
+   * Creates a new collection with the sorted versions.
+   *
+   * @return versions sorted by date
+   */
+  public Collection<ArtifactVersion> sortByReleaseDate() {
+    return ArtifactVersion.sortByReleaseDate(elements);
+  }
+
+  /**
+   * Sort artifact versions hold by ArtifactVersions by release date.
+   *
+   * @param artifactVersions the artifact versions
+   * @return sorted collection of ArtifactVersion
+   */
+  public static Collection<ArtifactVersion> sortByReleaseDate(
+      Value<ArtifactVersions> artifactVersions) {
+
+    if (artifactVersions.isUnknown()) {
+      return Collections.emptyList();
+    }
+    return ArtifactVersion.sortByReleaseDate(artifactVersions.get().elements);
+  }
+
+  /**
+   * Returns Artifact version with given version.
+   * If the version is not available an empty optional is returned.
+   *
+   * @param version to be searched version
+   * @return found version or empty optional
+   */
+  public Optional<ArtifactVersion> get(String version) {
+    return elements.stream()
+        .filter(v -> v.getVersion().equals(version))
+        .findFirst();
+  }
+
+  /**
+   * Get a set of versions.
+   *
+   * @return The versions.
+   */
+  @JsonGetter("elements")
+  public Set<ArtifactVersion> get() {
+    return new HashSet<>(elements);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o instanceof ArtifactVersions == false) {
+      return false;
+    }
+    ArtifactVersions other = (ArtifactVersions) o;
+    return Objects.equals(elements, other.elements);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(elements);
+  }
+
+  @Override
+  public String toString() {
+    if (elements.isEmpty()) {
+      return "No versions";
+    }
+    Collection<ArtifactVersion> sortByReleaseDate = sortByReleaseDate();
```

Comment:
Not sure how often `toString()` is called but maybe the sorting can happen in the constructor. You can then also get rid of `sortByReleaseDate()` if you return an unmodifiebale collection (or a clone).

Reply:
My first thought was to sort the artifact versions only on demand.
But currently every score use the sorted list 😉
Will update it as suggested.

Reply:
I see the constructor sorts elements now. Not sure if we need to call `sortByReleaseDate()` here. Not even sure if we need this method at all.